subworkflows/nf-core/vcf_gather_bcftools/main.nf

```diff
@@ -0,0 +1,51 @@
+include { BCFTOOLS_CONCAT    } from '../../../modules/nf-core/bcftools/concat/main'
+include { BCFTOOLS_SORT      } from '../../../modules/nf-core/bcftools/sort/main'
+include { TABIX_TABIX        } from '../../../modules/nf-core/tabix/tabix/main'
+
+workflow VCF_GATHER_BCFTOOLS {
+
+    take:
+    ch_vcfs             // channel: [ meta, vcf, tbi ]
+    ch_scatter_output   // channel: [ meta, bed, gather_count ] => output from the scatter subworkflow, if you didn't use this subworkflow you can just use `[]` as bed since it isn't used
+    common_meta         // string:  The name of the meta field that should become the new id
+    sort                // boolean: Whether or not the output file should be sorted !! Add the config when using sort !!
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    ch_vcfs
+        .join(ch_scatter_output)
+        .map(
+            { meta, vcf, tbi, bed, gather_count ->
+                meta = meta + [id:meta[common_meta]]
+                [ groupKey(meta, gather_count), vcf, tbi ]
+            }
+        )
+        .groupTuple()
+        .set { ch_concat_input }
```

Comment:
General nf-core consensus is we don't like `.set`, please replace with `ch_concact_input = ch_vcfs.join()` etc., everywhere

it makes it more readable in terms of identifying where each channel is created (when you're tracing back which channel goes where)

Reply:
Same comment thant @jfy133

Reply:
Allright, good to know lemme fix that :)

Reply:
Psh, @drpatelh is not (yet?) a god!

Reply:
Gotta keep the soon to be gods on my good side, right :grin: